packages/vite/src/node/server/pluginContainer.ts

```diff
@@ -960,14 +960,41 @@ class PluginContainer {
   }
 
   getModuleInfo(id: string): ModuleInfo | null {
-    return (
-      (
-        this.environments.client as DevEnvironment
-      ).pluginContainer.getModuleInfo(id) ||
-      (this.environments.ssr as DevEnvironment).pluginContainer.getModuleInfo(
-        id,
-      )
-    )
+    const clientModuleInfo = (
+      this.environments.client as DevEnvironment
+    ).pluginContainer.getModuleInfo(id)
+    const ssrModuleInfo = (
+      this.environments.ssr as DevEnvironment
+    ).pluginContainer.getModuleInfo(id)
+
+    if (clientModuleInfo == null && ssrModuleInfo == null) return null
+
+    return new Proxy({} as any, {
```

Comment:
This code here and below looks similar to the change in `mixedModuleGraph`. I initially also tried to refactor so we pass the mixed module graph to the plugin container so we can share the code, but it turns out a little fragile I think especially type-wise.

We also need to explicitly call the `getModuleInfo` here to trigger lazily creating `module.info` in `EnvironmentPluginContainer`.